### PR TITLE
feat: add `withSymlinks` argument back

### LINKED ISSUE #103: [Feature] add back crawling by symlink path, not real path
It looks like this feature was silently removed in #81, and now it always returns realpaths. cdb201f mentions replacing it with a new parameter to `withSymlinks`, but this was removed as well even though the documentation still mentions it

ref: #83

### PR DESCRIPTION
**Pull request by SuperchupuDev**
While working on `tinyglobby` I've noticed there is no way to disable `fdir` returning the real path of a symlink anymore, due to #81 silently removing support for it (even though the documentation still mentions the feature!).

This PR adds it back, inside an object parameter instead for better API design (see https://github.com/thecodrr/fdir/issues/83#issuecomment-1282787956). It defaults to `true` for backwards compatibility

Closes #103

### CODE REVIEW

**Comment by SuperchupuDev** [comment]
hi @thecodrr, would you mind taking a look at this anytime soon? symlinks wont quite work in my library until this (or an equivalent solution) is merged

**Comment by thecodrr** [comment]
Relevant commit: https://github.com/thecodrr/fdir/commit/cdb201f4f2c3afdf05f46e42894a8ef5836568a0

File: src/api/functions/resolve-symlink.ts
Comment (thecodrr) [inline src/api/functions/resolve-symlink.ts:21]: Let's use `fs.lstat` here as well.

File: src/api/functions/resolve-symlink.ts
Comment (thecodrr) [inline src/api/functions/resolve-symlink.ts:63]: Let's use `lstatSync` here.

**Comment by SuperchupuDev** [comment]
yes, that commit is what this PR aims to implement, but #81 removed the `withSymlinks` parameter that commit introduced. i can rename this pr to make that clearer

File: src/api/functions/resolve-symlink.ts
Comment (thecodrr) [inline src/api/functions/resolve-symlink.ts:94]: Let's invert these checks to be more consistent.

```suggestion
  if (options.useRealPaths)
    return isSynchronous
        ? resolveSymlinksWithRealPathsSync
        : resolveSymlinksWithRealPathsAsync;
  return isSynchronous
    ? resolveSymlinksSync
    : resolveSymlinksAsync;
```

**Review by thecodrr — CHANGES_REQUESTED** [review]
Thank you so much for adding this back!

**Comment by SuperchupuDev** [comment]
changing it to lstat makes this test fail as it's returning non-resolved symlinks
image

**Comment by SuperchupuDev** [comment]
@thecodrr i've reverted the `lstat` change, it makes behavior the same as if not using `withSymlinks` at all for some reason. not to worry though as it seems to work as intended with `stat`

**Comment by SuperchupuDev** [comment]
@thecodrr is there anything blocking the PR?

**Comment by thecodrr** [comment]
> @thecodrr i've reverted the lstat change, it makes behavior the same as if not using withSymlinks at all for some reason. not to worry though as it seems to work as intended with stat

I looked more into this and apparently, here's the correct way to go about this:

- When `resolvePaths` is `true`, we should use `realpath` and `stat`.
- When `resolvePaths` is `false`, we should use `lstat`.

`lstat` works directly on the symlink (i.e. it doesn't check if the symlink points to a file on disk) which is what we want when `resolvePaths` is `false`. Doing `lstat` on a dirent we "know" for sure isn't a symlink is just waste of resources.

Another thing I noticed while looking at this is that the tests checking for errors are just plain wrong (no idea why I added tests that way). They'll pass even if there's no error. Instead of testing for errors, let's just test the output and see if the symlink was resolved or not. That'll be more foolproof.

**Pull request merged**
